pkg/cli/admin/inspect/writer.go

```diff
@@ -51,7 +51,7 @@ func (r *resourceWriterReadCloser) Close() error {
 type simpleFileWriter struct{}
 
 func (f *simpleFileWriter) Write(filepath string, src fileWriterSource) error {
-	dest, err := os.OpenFile(filepath, os.O_RDWR|os.O_CREATE, 0755)
+	dest, err := os.OpenFile(filepath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
```

Comment:
In terms of yaml files generated by oc adm inspect command, this is indeed a problem and I'm supportive of this change. 

On the other hand, this writer is also used for saving the collected logs of pods, operators, etc. and I think, we should ensure that this change will not cause truncating the logs.

Reply:
Well this always starts writing from the beginning, so it will either overwrite from the beginning (causing the same bug) or drop the "old" data. The only way to behave differently would be to pass O_APPEND.

Or is there something I am missing?

Reply:
I agree with you @MarSik. Since we are not passing O_APPEND, we are already overwriting on the old data and it would be good to truncate it.

Thanks for fixing this.